src/python_tools/robot_broadcast_receiver.py

```diff
@@ -0,0 +1,53 @@
+from shared.proto.announcement_pb2 import Announcement
+import socket
+import argparse
+from time import time
+
+RECEIVE_TIMEOUT = 0.2
+
+
+def receive_announcements(port: int) -> [Announcement]:
+    """
+    Returns a list of Announcements, without duplicates received within a time window of 4s on a specified port
+    :param port: the port to listen for announcements on
+    :return: a list of Announcements, without duplicates
+    """
+    receiver = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    receiver.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+    receiver.settimeout(RECEIVE_TIMEOUT)
+    receiver.bind(("", port))
+
+    announcements = []
+    timeout = time() + 4  # 4s
```

Comment:
it would be nice to have the 4s as a constant at the top

Reply:
added